### DISCUSSION TITLE
Cropping An Uploaded Image

### DISCUSSION THREAD

**Discussion by IdanSheinkman** [yasirkula/UnityNativeGallery#290] [Q&A] [answered]
Is it possible to add a crop menu after picking the desired image from the phone's gallery? In my game the player is a square shape and the uploaded image is replacing him.

**Comment by yasirkula** [comment]
You can use Image Cropper plugin: https://github.com/yasirkula/UnityImageCropper

**Reply by IdanSheinkman** [reply]
Thanks!

**Reply by IdanSheinkman** [reply]
Do you have any example code of using the two plugins combined?
Specifically cropping an image that was uploaded from gallery.
Thank you very much for both of the plugins and the help.

**Reply by yasirkula** [reply, marked as answer]
```csharp
NativeGallery.GetImageFromGallery( ( path ) =>
{
	if( path != null )
	{
		Texture2D texture = NativeGallery.LoadImageAtPath( path, maxSize );
		if( texture == null )
		{
			Debug.Log( "Couldn't load texture from " + path );
			return;
		}
		
		ImageCropper.Instance.Show( texture, ( bool result, Texture originalImage, Texture2D croppedImage ) =>
		{
			if( result )
			{
				// Crop successful, use croppedImage as you wish
			}
			else
			{
				// Crop canceled
			}
		} );
	}
} );
```
